src/design/plone/contenttypes/browser/move_content/move_news_items.py

```diff
@@ -0,0 +1,79 @@
+from zope.schema.interfaces import IVocabularyFactory
+from zope.component import getUtility
+from plone import api
+from plone.memoize import ram
+from Products.Five import BrowserView
+
+from plone.app.contenttypes import _
+
+import time
+import logging
+import pkg_resources
+
+
+JS_TEMPLATE = (
+    "{portal_url}/++plone++design.plone.contenttypes/js/{name}.js?v={version}"  # noqa
+)
+
+logger = logging.getLogger(__name__)
+
+
+class View(BrowserView):
+    def __call__(self, *args, **kwargs):
+        self.move_data()
+        return super().__call__(*args, **kwargs)
+
+    def move_data(self):
+        if self.request.form.get("move", ""):
+            path = self.request.form.get("to_path")
+            if path:
+                move_to = api.content.get(path)
+                if not move_to:
+                    self.context.plone_utils.addPortalMessage(
+                        _("Indicated path is not valid"), "error"
+                    )
+                    return
+
+                for item_uid in [i for i, j in self.request.form.items() if j == "on"]:
```

Comment:
Anche qui, mi sembra che puoi evitare di fare un doppio ciclo for solo per estrarre gli uid.
Basta iterare su request.form.items e poi fare l'if come prima riga nel for

Reply:
Scrivo il commento sopra per spiegare cosa succede, che coss il codice è più laconico

Reply:
No, non mi piace. Il codice deve essere semplice da leggere e i for innestati non sono mai immediati da capire (il commento è barare).

Reply:
solved